### PR TITLE
Makefile improvements.

### PR DESCRIPTION
**Pull request by sagiegurari**
Really cool crate. Trying to help a bit sort out and remove duplications in the makefile.

* Default message is now in duckscript so its same code for all platforms. no need for code duplication.
* start/stop search task has a base task so common code is there.
* start/stop search task no longer has a windows specific implementation
* Move CI env var which isn't supported for all CI vendors to CARGO_MAKE_CI which supports all.

Few questions/comments

1. I couldn't test it properly. i'm working on gitpod and i can't run docker. would love some help testing this.
2. why did you have the windows specific overrides? 
i understand why you need those when using script (that's why i converted to command+args which support var value replacement) but for start/stop elasticsearch tasks i'm not sure why it was needed.
I don't have a windows machine to test (again need help with that as well) so i would love to understand more why your original tasks didn't work.
because it looks like for those you simply also defined STACK_VERSION and TEST_SUITE on the command, but those env vars are already set.

### CODE REVIEW

**Review by russcam — CHANGES_REQUESTED** [review]
Thanks for opening @sagiegurari, and thank you also for cargo make, it's exactly what I was looking for 😄 I've left some comments.

Would you mind [signing the CLA](https://www.elastic.co/contributor-agreement), so that this PR could eventually be pulled in?

File: Makefile.toml [hunk 1]
```diff
@@ -77,33 +75,30 @@ script = ["cargo install cargo2junit"]
...
-condition = { env_set = [ "CI" ] }
+condition = { env_true = [ "CARGO_MAKE_CI" ] }
 script = ["cat test_results/results.json | cargo2junit > test_results/cargo-junit.xml"]
 dependencies = ["install-cargo2junit"]
 
+[tasks.run-elasticsearch]
+category = "Elasticsearch"
+condition = { env_set = [ "STACK_VERSION", "TEST_SUITE" ], env_false = ["CARGO_MAKE_CI"] }
+script_runner = "bash"
+script = { file = "./.ci/run-elasticsearch.sh" }
```
[the remark this answers is not in the document]
Comment (sagiegurari) [reply Makefile.toml:86]: thanks for the info. i'll revert this one, or think of another way.

File: Makefile.toml [hunk 2]
```diff
@@ -28,24 +28,22 @@ Runs yaml_test_runner package to generate tests from yaml files for a given Elas
...
 dependencies = ["start-elasticsearch"]
 
-[tasks.run-yaml-test-runner.windows]
-script = ["cargo run -p yaml_test_runner -- -u %ELASTICSEARCH_URL%"]
-
 [tasks.test-yaml-test-runner]
 category = "Elasticsearch"
 private = true
-condition = { env_set = [ "ELASTICSEARCH_URL" ], env_not_set = ["CI"] }
+condition = { env_set = [ "ELASTICSEARCH_URL" ], env_false = ["CARGO_MAKE_CI"] }
```
[the remark this answers is not in the document]
Comment (sagiegurari) [reply Makefile.toml:38]: no need to touch that shell. ci_info crate will detect its CI env if you did CI=true.
than cargo make will set CARGO_MAKE_CI=true
so its fine.
the advantage of this, is that you can later on support other CI providers that do not set the CI env var.

File: Makefile.toml [hunk 3]
```diff
@@ -142,42 +141,22 @@ run_task = "format"
...
     echo - test-generator: Generates and runs api_generator package tests
     echo - test: Runs elasticsearch package tests against a given Elasticsearch version
-    echo.
+    echo
     echo Most tasks use these environment variables:
     echo - STACK_VERSION (default '$STACK_VERSION'): the version of Elasticsearch
     echo - TEST_SUITE ('oss' or 'xpack', default '$TEST_SUITE'): the distribution of Elasticsearch
-    echo - CI (default not set): set when running on CI to determine whether to start Elasticsearch and format test output as JSON
-    echo.
+    echo - CARGO_MAKE_CI (default not set): set when running on CI to determine whether to start Elasticsearch and format test output as JSON
```
Comment (sagiegurari) [inline Makefile.toml:158]: actually this should stay CI and not CARGO_MAKE_CI
its automatically set for you based on the env.

File: Makefile.toml [hunk 4]
```diff
@@ -142,42 +137,22 @@ run_task = "format"
...
     echo - stop-elasticsearch: Stops Elasticsearch docker container, if running
     echo - test-yaml: Generates and runs yaml_test_runner package xpack/oss tests against a given Elasticsearch version
     echo - test-generator: Generates and runs api_generator package tests
     echo - test: Runs elasticsearch package tests against a given Elasticsearch version
-    echo.
+    echo
     echo Most tasks use these environment variables:
     echo - STACK_VERSION (default '$STACK_VERSION'): the version of Elasticsearch
     echo - TEST_SUITE ('oss' or 'xpack', default '$TEST_SUITE'): the distribution of Elasticsearch
     echo - CI (default not set): set when running on CI to determine whether to start Elasticsearch and format test output as JSON
```
[the remark this answers is not in the document]
Comment (sagiegurari) [reply Makefile.toml:162]: sorry, keeping it CI as CARGO_MAKE_CI is automatically defined for you by cargo-make internally with true/false value.

**Comment by sagiegurari** [comment]
* signed the CLA
* few updates per comments (double check me)
* merged changes
* improved instructions printing per platform after merges

**Review by russcam — CHANGES_REQUESTED** [review]
Thanks @sagiegurari, I just left one small suggestion but otherwise LGTM 👍  Thank you for your conribution!

File: Makefile.toml [hunk 5]
```diff
@@ -197,57 +200,35 @@ cargo package %{CARGO_MAKE_CARGO_PACKAGE_FLAGS}
...
     echo Most tasks use these environment variables:
-    echo - STACK_VERSION (default '%STACK_VERSION%'): the version of Elasticsearch
-    echo - TEST_SUITE ('oss' or 'xpack', default '%TEST_SUITE%'): the distribution of Elasticsearch
+    if is_windows
+        echo - STACK_VERSION (default '%STACK_VERSION%'): the version of Elasticsearch
+        echo - TEST_SUITE ('oss' or 'xpack', default '%TEST_SUITE%'): the distribution of Elasticsearch
+    else
+        echo - STACK_VERSION (default '$STACK_VERSION'): the version of Elasticsearch
+        echo - TEST_SUITE ('oss' or 'xpack', default '$TEST_SUITE'): the distribution of Elasticsearch
+    end
```
[the remark this answers is not in the document]
Comment (sagiegurari) [reply Makefile.toml:229]: oh ok, so you want the actual value. for some reason (my mistake) i thought you wanted to show what env var it is :)
i'll fix that.

[the remark this answers is not in the document]
Comment (russcam) [reply Makefile.toml:229]: Yeah, the actual value is what we want. Sorry for any confusion 😄

**Comment by sagiegurari** [comment]
@russcam I think everything is handled, but can you double check?

**Review by russcam — APPROVED** [review]
LGTM, thanks @sagiegurari 👍

**Pull request merged**